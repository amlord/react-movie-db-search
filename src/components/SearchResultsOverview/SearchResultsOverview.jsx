import React from 'react';
import PropTypes from 'prop-types';

const SearchResultsOverview = ({ totalResults }) => (
  <div>
    <span>Total Results: </span>
    <strong>{totalResults}</strong>
  </div>
);

SearchResultsOverview.propTypes = {
  totalResults: PropTypes.number.isRequired,
};

export default SearchResultsOverview;
