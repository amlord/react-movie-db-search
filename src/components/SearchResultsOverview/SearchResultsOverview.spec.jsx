import React from 'react';
import { shallow } from 'enzyme';
import SearchResultsOverview from './SearchResultsOverview';

describe('SearchResultsOverview', () => {
  it('displays correctly', () => {
    const wrapper = shallow(<SearchResultsOverview />);
    expect(wrapper).toMatchSnapshot();
  });
});
