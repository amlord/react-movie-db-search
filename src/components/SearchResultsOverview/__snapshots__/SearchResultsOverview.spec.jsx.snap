// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsOverview displays correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResultsOverview
    totalResults={12}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          Total Results: 
        </span>,
        <strong>
          12
        </strong>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Total Results: ",
        },
        "ref": null,
        "rendered": "Total Results: ",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 12,
        },
        "ref": null,
        "rendered": 12,
        "type": "strong",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            Total Results: 
          </span>,
          <strong>
            12
          </strong>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Total Results: ",
          },
          "ref": null,
          "rendered": "Total Results: ",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 12,
          },
          "ref": null,
          "rendered": 12,
          "type": "strong",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
